src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/NeuralNetInference.java

```diff
@@ -0,0 +1,291 @@
+ * gatk NeuralNetInference\
+ *   -V vcf_to_annotate.vcf.gz \
+ *   -R reference.fasta \
+ *   -O annotated.vcf \
+ *   --architecture src/main/resources/org/broadinstitute/hellbender/tools/walkers/vqsr/cnn_1d_annotations.hd5
+ * </pre>
+ *
+ */
+@ExperimentalFeature
+@CommandLineProgramProperties(
+        summary = NeuralNetInference.USAGE_SUMMARY,
+        oneLineSummary = NeuralNetInference.USAGE_ONE_LINE_SUMMARY,
+        programGroup = VariantEvaluationProgramGroup.class
+)
+
+public class NeuralNetInference extends VariantWalker {
+    private final static String NL = String.format("%n");
+    static final String USAGE_ONE_LINE_SUMMARY = "Apply 1d Convolutional Neural Net to filter annotated variants";
+    static final String USAGE_SUMMARY = "Annotate a VCF with scores from 1d Convolutional Neural Network (CNN)." +
+            "The CNN will look at the reference sequence and variant annotations to determine a Log Odds Score for each variant.";
+
+    private static final int CONTIG_INDEX = 0;
+    private static final int POS_INDEX = 1;
+    private static final int REF_INDEX = 2;
+    private static final int ALT_INDEX = 3;
+    private static final int KEY_INDEX = 4;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc = "Output file")
+    private String outputFile = null; // output file produced by Python code
+
+    @Argument(fullName = "architecture", shortName = "a", doc = "Neural Net architecture and weights hd5 file", optional = false)
+    private String architecture = null;
+
+    @Argument(fullName = "window-size", shortName = "ws", doc = "Neural Net input window size", minValue = 0, optional = true)
+    private int windowSize = 128;
+
+    @Advanced
+    @Argument(fullName = "inference-batch-size", shortName = "ibs", doc = "Size of batches for python to do inference on.", minValue = 1, maxValue=4096, optional = true)
+    private int inferenceBatchSize = 256;
+
+    @Advanced
+    @Argument(fullName = "transfer-batch-size", shortName = "tbs", doc = "Size of data to queue for python streaming.", minValue = 1, maxValue=8192, optional = true)
+    private int transferBatchSize = 512;
+
+    @Hidden
+    @Argument(fullName = "enable-journal", shortName = "journal", doc = "Enable streaming process journal.", optional = true)
+    private boolean enableJournal = false;
+
+    @Hidden
+    @Argument(fullName = "keep-temp-file", shortName = "ktf", doc = "Keep the temporary file that python writes scores to.", optional = true)
+    private boolean keepTempFile = false;
+
+    // Create the Python executor. This doesn't actually start the Python process, but verifies that
+    // the requestedPython executable exists and can be located.
+    final StreamingPythonScriptExecutor pythonExecutor = new StreamingPythonScriptExecutor(true);
+
+    private FileOutputStream fifoWriter;
+    private VariantContextWriter vcfWriter;
```

Comment:
This scope of this variable can be reduced now. See comment below.

Reply:
fixed